esx_service/vmdk_ops.py

```diff
@@ -506,9 +506,13 @@ def vol_info(vol_meta, vol_size_info, datastore):
     vinfo[LOCATION] = datastore
 
     if kv.ATTACHED_VM_UUID in vol_meta:
-       vm = findVmByUuid(vol_meta[kv.ATTACHED_VM_UUID])
-       if vm:
-          vinfo[ATTACHED_TO_VM] = vm.config.name
+        vm = findVmByUuid(vol_meta[kv.ATTACHED_VM_UUID])
+        if vm:
+            vinfo[ATTACHED_TO_VM] = vm.config.name
+        elif vol_meta[kv.ATTACHED_VM_NAME]:
```

Comment:
Again check if kv.ATTACHED_VM_NAME is in vol_meta before indexing vol_meta with it,